src/routes/sendEvent.ts

```diff
@@ -0,0 +1,29 @@
+import { Router } from 'express';
+import { nanoid } from 'nanoid';
+import { eventConsumer } from '../eventConsumer';
+
+const router = Router();
+
+//todo: validate schema
+router.post('/', (req, res) => {
+  console.log(`received messageBody -> ${JSON.stringify(req.body)}`);
+  const requestId = req.body.id ?? nanoid();
+  const detailType = req.body['detail-type'];
+
+  if (eventConsumer[detailType] == null) {
+    throw new Error(
+      `Unable to retrieve handler for detailType='${detailType}'`
+    );
+  }
+
+  eventConsumer[detailType](req.body);
```

Comment:
this will be an async call running in background..i couldn't await this inside the `post` call

is there a way to `await` them inside the call?

we can expect this to execute it in the background but if we can await, we can throw 5xx and store messages in the lambda DLQ

Reply:
Yep, you need to add async to the function definition

e.g. `router.post('/', async (req, res) => {`

Reply:
thanks! done